### ISSUE TITLE
Add reserve factor to LH

### ISSUE THREAD

**Issue by miguelmtzinf** [aave/aave-v4#39]
- The DAO takes a cut from the yield payed by borrowers
- This cut can be taken fin form of shares.

(we can check how Morpho does it, it must be pretty similar)

**Comment by EthWarrior** [comment]
Could this be implemented as a strategy? i.e. would open opportunities to explore dynamic reserve factor even (reserve factor could increase and decrease based on market conditions). Charge more when demand is overly high.

**Comment by CheyenneAtapour** [comment]
@EthWarrior Yeah, I think that is possible. We'll build the basic case (constant RF) first then explore that

**Comment by DhairyaSethi** [comment]
currently the best approach is to maintain an accruedToTreasury counter and subtract from totalAssets for supply exchange ratio

**Comment by EthWarrior** [comment]
Actually I just realized that Compound V3 has dynamic reserve factor but its somehow not optimal as it needs constant regulation. It could work with a PID but my concern is that it might be overengineered and hence doesn't beat the good old fixed RF.

**Comment by miguelmtzinf** [comment]
Some notes here: https://www.notion.so/aave/Notes-on-Reserve-Factor-implementation-1979d63a22de8020a32ed01fe7445599

**Comment by CanonicalJP** [comment]
Synced with Emilio, Miguel, and Dhairya to explore 2 different strategies:
1. Have custom logic in the LH but the Spokes don't know about it
2. Have a treasury Spoke and new logic in the LH only to set the address (and mint shares)

Notes:
- Agreed that the 2nd alternative looks like the best option to avoid having custom logic in LH and keep it simple
- Need to mint shares to treasury Spoke to account for reserve factor
- The treasury Spoke will be subject to the same restrictions as any other spoke (e.g. reserve freeze). This is how it’s implemented in V3 nowadays. It's OK.
- Don’t need to account for the reserve factor in the interest rate strategy, to be validated
- If the amount is less than 1 share, it’s donated. Don't see any issues, but will explore edge cases?
- The treasury Spoke can have custom logic to include more complex functionalities (e.g. more than a treasury)
- Having the treasury Spoke upgradeable wouldn't be terrible cause it's not user facing. However, it'd be better to just make it immutable, deploy a new treasury Spoke, and update the address in the LH, if needed
- There's room for exploration on how to set the address in the LH, like making the treasury Spoke the id 0 Spoke of every LH. 
- Logic in the treasury Spoke should be as generic as possible to facilitate other functionalities like integration with liquidations

Next Steps:
- Implement treasury Spoke
- Add logic to set the address of the treasury in the LH
- Add logic to mint shares to the treasury Spoke in the LH
- Test for any edge cases

**Comment by yan-man** [comment]
> Add logic to set the address of the treasury in the LH

- are we going to enforce this into the constructor?

do we want to have a custom RF per spoke? if not we can close #190
